### ISSUE TITLE
Install never passes Deactivating and disabling dhcpcd

### ISSUE THREAD

**Issue by koosthoe** [resin-io/resin-wifi-connect#194]
I did this command since my install was auto-canceling.
`bash <(curl -L https://github.com/resin-io/resin-wifi-connect/raw/master/scripts/raspbian-install.sh)`

It starts out fine but then times out at this part,
WiFi Connect Raspbian Installer: Deactivating and disabling dhcpcd...

any ideas how to fix? im using raspbian stretch lite on a RPI Zero W

**Comment by majorz** [comment]
I do not have a Zero W, but I will check it out on a 1B+. Hopefully I will be able to reproduce, since it is the same Broadcom BCM2835 SOC.

**Comment by majorz** [comment]
I checked RPi1 B+ - no problems there. Are you accessing the device with ssh through a wireless network, or you are using a keyboard and monitor attached to the device? If it is through a wireless network only, then when dhcpcd is disabled, the ssh connection will drop and you will lose connectivity with the device.

**Comment by koosthoe** [comment]
Yes I am accessing with ssh. I do not have a keyboard and monitor for the device. Is there anyway to do this without a keyboard and monitor for the device through USB? I am fairly new to RPI.

**Comment by majorz** [comment]
You can run the install command with appended `&& wifi-connect`:

`bash <(curl -L https://github.com/resin-io/resin-wifi-connect/raw/master/scripts/raspbian-install.sh) && wifi-connect`

This will download and install WiFi Connect and then it will run it. The application will create an access point to which you may connect with your mobile phone. There you may setup the wireless credentials again and then it will reconnect you to the wireless network.

Please let me know how would that work for you.

**Comment by koosthoe** [comment]
Tried this but I am getting the same results as before.

**Comment by majorz** [comment]
Right, when the ssh session closed, the install process will get terminated as well. The command has to be executed in the background. Let me try this running it locally to make sure it is all running well and I will get back to you with results.

**Comment by majorz** [comment]
This is how I did it:

* ssh into the RPi: `ssh pi@192.168.1.129`, where `192.168.1.129` was the IP of my RPi. 
* Create `install.sh` in the home folder of the pi user with the following content:
```
#!/usr/bin/env bash
bash <(curl -L https://github.com/resin-io/resin-wifi-connect/raw/master/scripts/raspbian-install.sh) -- -y
wifi-connect
```
* Make it executable with `chmod +x install.sh`
* Exit the ssh session
* Run from the terminal of my laptop: `ssh pi@192.168.1.129 "nohup sudo ~/install.sh >> ~/install.log 2>&1 &"`
  - This will run the `install.sh` script with root permissions in the background.
  - In addition it will redirect all output to an `install.log` file in the home folder.
* Wait for a while for the installation process to complete - this will take minutes.
* Connected to WiFi Connect with my phone.

**Comment by majorz** [comment]
I will mark this ticket as documentation issue, as we need to add a warning for the installation not to happen on top of a ssh session through a wireless connection. The above instructions would work, but this is not something I would generally recommend, as there is no visible output of how the command is running.

I think the proper way for installing this on RPi Zero W with Raspbian should be preloading our application after the SD card is burnt, e.g. with [Etcher](https://etcher.io/). This will need more investigation though.

**Comment by koosthoe** [comment]
Yeah im still having issues. I tried this multiple times and couldnt get the wifi-connect to show up. Also cant see what the possible errors are with this method.

**Comment by majorz** [comment]
It is a Catch-22 situation with installing this on RPi Zero W with Raspbian without a keyboard and monitor attached.

This is one of those situations where you will get much better results if you use our own resinOS. You may check our the Zero W getting started docs here: https://docs.resin.io/learn/getting-started/raspberry-pi/nodejs/

**Comment by koosthoe** [comment]
I got the RPI zero W connected to a keyboard and monitor. Now the install stops with this error

command failed: curl https://api.github.com/repos/resin-io/resin-wifi-connect/releases/latest -s

**Comment by majorz** [comment]
@koosthoe 

Hi, very sorry for getting a bit late to you on this. I was on a very busy support shifts in the last couple of days. I will make a new branch and send you a link with a new install script that fixes this.

Basically we need to download and install WiFi Connect before enabling NetworkManager, since after NetworkManager is installed on Zero W, there won't be available Internet connection.

**Comment by majorz** [comment]
@koosthoe Before I PR this and merge it, can please you try:
```
bash <(curl -L https://github.com/resin-io/resin-wifi-connect/raw/194-raspbian-zero-w/scripts/raspbian-install.sh)
```

This should fix the issue you experienced.

**Comment by koosthoe** [comment]
This seemed to install fully but I am not getting a wifi-connect hotspot to show up on my phone or computer

**Comment by majorz** [comment]
This only installs it. You need to run it next with `sudo wifi-connect`. Please let me know if this was the issue.

**Comment by koosthoe** [comment]
YES! its working now. thanks so much for the help. Much appreciated

**Comment by majorz** [comment]
@charliesneath the Raspbian install script does not run wifi-connect after it is installed on the system. You may wrap the installer invocation call + starting wifi-connect in a  `nohup &` call, so that the command continues to run after the SSH connection is terminated. Could you try that and let me know your findings? And let me know if you have additional questions.

**Comment by charliesneath** [comment]
@majorz Can you be more specific and show what the command should be? I see the examples above, but I’m newer to working with Linux and the example would be very helpful. Thank you!

**Comment by majorz** [comment]
@charliesneath You can create a file called `install-wifi-connect.sh`:
```
bash <(curl -L https://github.com/balena-io/wifi-connect/raw/master/scripts/raspbian-install.sh) -- -y
sudo wifi-connect
```

Set permissions to executable: `chmod +x install-wifi-connect.sh`

And then run `nohup bash ./install-wifi-connect.sh & tail -f nohup.out`

**Comment by majorz** [comment]
@charliesneath `tail` tries to read too quickly. You may fix this by using `-F` instead of `-f`:

`nohup bash ./install-wifi-connect.sh & tail -F nohup.out`

**Comment by charliesneath** [comment]
@majorz Thanks! Looks it's working now. Appreciate the quick replies here.
